src/DssSpell.t.base.sol

```diff
@@ -2370,4 +2540,228 @@ contract DssSpellTestBase is Config, DssTest {
+            uint256 preserveSky = sky.balanceOf(address(pair));
+
+            uint256 pbalanceUsdsFarm;
+            // Checking the farm balance is only relevant if split.burn() < 100%
+            if (split.burn() < 1 * WAD) {
+                pbalanceUsdsFarm = usds.balanceOf(split.farm());
+                assertFalse(split.farm() == address(0), "TestError/Splitter/missing-farm");
+            }
+
+            vow.flap();
+
+            assertGt(pair.balanceOf(pauseProxy),      pbalancePauseProxy,   "TestError/Flapper/pair-pauseProxy-balance-no-increase");
+            assertGt(usds.balanceOf(address(pair)),   preserveUsds,         "TestError/Flapper/usds-pair-balance-no-increase");
+            assertEq(sky.balanceOf(address(pair)),    preserveSky,          "TestError/Flapper/unexpected-sky-pair-balance-change");
+            assertGt(pdaiVow - vat.dai(address(vow)), vow.bump() * 9 / 10,  "TestError/Flapper/vat-dai-vow-change-too-low");
+            assertLt(pdaiVow - vat.dai(address(vow)), vow.bump() * 11 / 10, "TestError/Flapper/vat-dai-vow-change-too-high");
+            assertEq(usds.balanceOf(address(flap)),   0,                    "TestError/Flapper/invalid-usds-balance");
+            assertEq(sky.balanceOf(address(flap)),    0,                    "TestError/Flapper/invalid-sky-balance");
+
+            if (split.burn() < 1 * WAD) {
+                assertEq(usds.balanceOf(split.farm()), pbalanceUsdsFarm + payWad, "TestError/Splitter/invalid-farm-balance");
+            }
+        }
+
+        // Check Mom can increase hop
+        {
+            // The check for the configured value is already done in `_checkSystemValues()`
+            assertLt(split.hop(), type(uint256).max);
+            vm.prank(chief.hat());
+            splitterMom.stop();
+            assertEq(split.hop(), type(uint256).max);
+        }
+    }
+
+    function _testSystemTokens() internal {
+        _vote(address(spell));
+        _scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
+
+        // USDS
+        {
+            // USDS is upgradeable, so we need to ensure the implementation contract address is correct.
+            assertEq(_imp(address(usds)), addr.addr("USDS_IMP"), "TestError/USDS/invalid-usds-implementation");
+        }
+
+        // Converter: Dai <-> USDS
+        {
+            DaiUsdsLike daiUsds = DaiUsdsLike(addr.addr("DAI_USDS"));
+            address daiHolder = address(0x42);
+            deal(address(dai), daiHolder, 1_000 * WAD);
+            address usdsHolder = address(0x65);
+            deal(address(usds), usdsHolder, 1_000 * WAD);
+
+
+            // Dai -> USDS conversion
+            {
+                uint256 before = vm.snapshot();
+
+                uint256 pdaiBalance  = dai.balanceOf(daiHolder);
+                uint256 pusdsBalance = usds.balanceOf(usdsHolder);
+
+                vm.startPrank(daiHolder);
+                dai.approve(address(daiUsds),  type(uint256).max);
```

Comment:
The four operations for dai<->usds/mkr<->sky have both tokens being approved when only the transferred one is needed.

Reply:
Fixed.